server/etcdserver/txn/txn.go

```diff
@@ -362,10 +343,11 @@ func executeTxn(ctx context.Context, lg *zap.Logger, txnWrite mvcc.TxnWrite, rt
 				traceutil.Field{Key: "req_type", Value: "put"},
 				traceutil.Field{Key: "key", Value: string(tv.RequestPut.Key)},
 				traceutil.Field{Key: "req_size", Value: tv.RequestPut.Size()})
-			resp, err := put(ctx, txnWrite, tv.RequestPut)
+			prevKV, err := getPrevKV(trace, txnWrite, tv.RequestPut)
 			if err != nil {
 				return 0, fmt.Errorf("applyTxn: failed Put: %w", err)
```

Comment:
should we change error message to `applyTxn: failed to get prev kv when put` or something?